compiler/rustc_expand/src/mbe/transcribe.rs

```diff
@@ -385,6 +422,94 @@ fn lockstep_iter_size(
                 _ => LockstepIterSize::Unconstrained,
             }
         }
+        TokenTree::MetaVarExpr(_, ref expr) => {
+            expr.ident().into_iter().fold(LockstepIterSize::Unconstrained, |size, name| {
+                let name = MacroRulesNormalizedIdent::new(name.clone());
+                match lookup_cur_matched(name, interpolations, repeats) {
+                    Some(matched) => match matched {
+                        MatchedNonterminal(_) => size,
+                        MatchedSeq(ref ads) => {
+                            size.with(LockstepIterSize::Constraint(ads.len(), name))
+                        }
+                    },
+                    _ => size,
+                }
+            })
+        }
         TokenTree::Token(..) => LockstepIterSize::Unconstrained,
     }
 }
+
+fn transcribe_metavar_expr<'a>(
+    cx: &ExtCtxt<'a>,
+    sp: &DelimSpan,
+    marker: &mut Marker,
+    expr: mbe::MetaVarExpr,
+    interp: &FxHashMap<MacroRulesNormalizedIdent, NamedMatch>,
+    repeats: &[(usize, usize)],
+    result: &mut Vec<TreeAndSpacing>,
+) -> PResult<'a, ()> {
+    let no_repetition_err = || Err(cx.struct_span_err(sp.entire(), "no repetitions at this depth"));
+    match expr {
+        mbe::MetaVarExpr::Ignore(..) => {}
+        mbe::MetaVarExpr::Count(original_ident, depth) => {
+            let ident = MacroRulesNormalizedIdent::new(original_ident);
+            match interp.get(&ident) {
+                Some(matched) => match count_repetitions(matched, &repeats, depth) {
+                    Some(count) => {
+                        marker.visit_span(&mut sp.entire());
```

Comment:
What does this line mean?
(And other `marker.visit_span(&mut sp.entire())` lines below.)

Reply:
Heh... I am not sure... :)

They do give the span an additional context but it seems that this additional context is not necessary. Therefore, I removed all `marker.visit_span`.